x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/DataFrameAnalysisConfig.java

```diff
@@ -32,6 +33,10 @@ public DataFrameAnalysisConfig(Map<String, Object> config) {
         }
     }
 
+    public DataFrameAnalysisConfig(DataFrameAnalysisConfig config) {
+        this.config = new HashMap<>(config.config);
```

Comment:
Shall we also use `this.config = Collections.unmodifiableMap(config.config)`?

Reply:
To avoid the nested wrapping problem with `Collections.unmodifiableMap` we would have to use `this.config = Collections.unmodifiableMap(new HashMap<>(config.config))`.

However, it would be better to use `Collections.unmodifiableMap` in the two other constructors instead, and then this one can simply copy the reference safe in the knowledge that the other map is unmodifiable.  If we make the class immutable like this then there's actually no need for this copy constructor at all...